pageserver/src/tenant/timeline.rs

```diff
@@ -4493,11 +4629,18 @@ impl<'a> TimelineWriter<'a> {
         let initial_size = open_layer.size().await?;
         let last_freeze_at = self.last_freeze_at.load();
         let last_freeze_ts = *self.last_freeze_ts.read().unwrap();
+
+        let writer_state = self.write_guard.take();
```

Comment:
We bail out at the top of the function if there's state in the write guard already. Why handle that case here?

Reply:
This is gone now.